api/base/serializers.py

```diff
@@ -758,6 +758,34 @@ def is_valid(self, clean_html=True, **kwargs):
 
         return ret
 
+class JSONAPIRelationshipSerializer(ser.Serializer):
+    """Base Relationship serializer. Requires that a `type_` option is set on `class Meta`.
+    Provides a simplified serialization of the relationship, allowing for simple update request
+    bodies. Also provides links to itself and to the endpoint providing information about the related
+    object.
+    """
+
+    def to_representation(self, obj, envelope='data'):
+        ret = {}
+        meta = getattr(self, 'Meta', None)
+        type_ = getattr(meta, 'type_', None)
+        assert type_ is not None, 'Must define Meta.type_'
+
+        data = collections.OrderedDict([
+            ('links', collections.OrderedDict()),
+            ('data', collections.OrderedDict()),
+        ])
+        relation_id_field = self.fields['id']
+
+        attribute = relation_id_field.get_attribute(obj)
+        relationship = relation_id_field.to_representation(attribute)
+
+        data['data'] = {'type': type_, 'id': relationship} if relationship else None
+        data['links'] = {key: val for key, val in self.fields.get('links').to_representation(obj).iteritems()}
+
+        ret.update({envelope: data} if envelope else data)
+        return ret
```

Comment:
Return `data` instead of `ret`.  Data and links should not need to be nested inside of `data`.  They should both be top-level keys.  http://jsonapi.org/format/#fetching-relationships

![screen shot 2015-12-14 at 11 32 44 am](https://cloud.githubusercontent.com/assets/9755598/11787397/7c9a3a30-a259-11e5-97af-b87cdd3d3c68.png)